dbt_subprojects/hourly_spellbook/models/_sector/perpetual/projects/avantis/avantis_v1_base_perpetual_trades.sql

```diff
@@ -0,0 +1,208 @@
+            WHEN 13 THEN 'LTC-USD'
+            WHEN 14 THEN 'MANA-USD'
+            WHEN 15 THEN 'OMG-USD'
+            WHEN 16 THEN 'SNX-USD'
+            WHEN 17 THEN 'UNI-USD'
+            WHEN 18 THEN 'XLM-USD'
+            WHEN 19 THEN 'XRP-USD'
+            WHEN 20 THEN 'ZEC-USD'
+            WHEN 31 THEN 'LUNA-USD'
+            WHEN 32 THEN 'YFI-USD'
+            WHEN 33 THEN 'SOL-USD'
+            -- Add more trading pairs as needed
+            ELSE CONCAT('pair_index_', CAST(pairIndex as VARCHAR))
+        END AS underlying_asset,
+
+        CASE pairIndex
+            WHEN 0 THEN 'BTC-USD'
+            WHEN 1 THEN 'ETH-USD'
+            WHEN 2 THEN 'LINK-USD'
+            WHEN 3 THEN 'DOGE-USD'
+            WHEN 4 THEN 'MATIC-USD'
+            WHEN 5 THEN 'ADA-USD'
+            WHEN 6 THEN 'SUSHI-USD'
+            WHEN 7 THEN 'AAVE-USD'
+            WHEN 8 THEN 'ALGO-USD'
+            WHEN 9 THEN 'BAT-USD'
+            WHEN 10 THEN 'COMP-USD'
+            WHEN 11 THEN 'DOT-USD'
+            WHEN 12 THEN 'EOS-USD'
+            WHEN 13 THEN 'LTC-USD'
+            WHEN 14 THEN 'MANA-USD'
+            WHEN 15 THEN 'OMG-USD'
+            WHEN 16 THEN 'SNX-USD'
+            WHEN 17 THEN 'UNI-USD'
+            WHEN 18 THEN 'XLM-USD'
+            WHEN 19 THEN 'XRP-USD'
+            WHEN 20 THEN 'ZEC-USD'
+            WHEN 31 THEN 'LUNA-USD'
+            WHEN 32 THEN 'YFI-USD'
+            WHEN 33 THEN 'SOL-USD'
+            -- Add more trading pairs as needed
+            ELSE CONCAT('pair_index_', CAST(pairIndex as VARCHAR))
+        END AS market,
+
+        contract_address AS market_address,
+        COALESCE(price_usd * position_size, position_size) AS volume_usd,
+        fee_usd,
+        position_size AS margin_usd,
+
+        CASE 
+            WHEN event_type = 'limit_order' THEN
+                CASE WHEN isBuy THEN 'long' ELSE 'short' END
+            WHEN event_type = 'margin_update' THEN
+                CASE WHEN isBuy THEN 'update_long' ELSE 'update_short' END
+        END AS trade,
+
+        'avantis' AS project,
+        '1' AS version,
+        'avantis' AS frontend,
+        trader,
+        COALESCE(price_usd * position_size, position_size) AS volume_raw,
+        evt_tx_hash AS tx_hash,
+        evt_index
+    FROM trading_events
+)
+
+SELECT
+    'base' AS blockchain,
+    CAST(date_trunc('DAY', perps.block_time) AS date) AS block_date,
+    CAST(date_trunc('MONTH', perps.block_time) AS date) AS block_month,
+    perps.block_time,
+    perps.virtual_asset,
+    perps.underlying_asset,
+    perps.market,
+    perps.market_address,
+    perps.volume_usd,
+    perps.fee_usd,
+    perps.margin_usd,
+    perps.trade,
+    perps.project,
+    perps.version,
+    perps.frontend,
+    perps.trader,
+    CAST(perps.volume_raw as UINT256) as volume_raw,
+    perps.tx_hash,
+    tx."from" AS tx_from,
+    tx."to" AS tx_to,
+    perps.evt_index
+FROM perps
+INNER JOIN {{ source('base', 'transactions') }} AS tx
```

Comment:
We need a incremental filter for all `transactions` tables.
We can make a transactions CTE with incremental first before inner join for faster loading.
i.e.
```
transactions_filtered AS (
    SELECT
      hash,
      block_number,
      "from",
      "to"
    FROM 
      {{ source('base','transactions') }}
    WHERE
      {% if is_incremental() %}
      {{ incremental_predicate('block_time') }}
      {% else %}
      block_time >= TIMESTAMP '{{project_start_date}}'
      {% endif %}
  ),
```